### ISSUE TITLE
SCM/CI Workflow integration fails to get workflow file for private GitLab repositories

### ISSUE THREAD

**Issue by nickbroon** [openSUSE/open-build-service#14079]
# Issue Description

When my private OBS instance receives webtriggers from GitLab.com for a private repository I see errors similar to:

```
<status code="non_existent_workflows_file">
  <summary>.obs/workflows.yml could not be downloaded from the SCM branch/commit abcd1234.
Is the configuration file in the expected place? Check https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html#sec.obs.obs_scm_ci_workflow_integration.setup.obs_workflows
403 Forbidden</summary>
</status>
```

And I knew that I definitely had `debian/obs/workflows.yml` file configured and the default branch for the repository.

I couldn't find any specific logs in `/srv/obs/log/` related to this (suggestions for logs to look at and/or config/debug to enable?).

But with some code inspection I spotted:

https://github.com/openSUSE/open-build-service/blob/3188b45aa94d6ce74af1dbcece107353fd2c34aa/src/api/app/services/workflows/yaml_downloader.rb#L38

```
      when 'gitlab'
        # This GitLab URL admits both a branch name and a commit sha.
        "#{@scm_payload[:api_endpoint]}/#{@scm_payload[:path_with_namespace]}/-/raw/#{@scm_payload[:target_branch]}/#{@token.workflow_configuration_path}"
```

This just appears to be using direct a URL to raw public files in a GitLab repository, not the actual GitLab file access API that uses a token and can work with any repository (See: https://docs.gitlab.com/ee/api/repository_files.html#get-raw-file-from-repository). ie something like:

```
curl --header "PRIVATE-TOKEN: <access_token>" "https://gitlab.com/api/v4/projects/1234/repository/files/%2Eobs%2Fworkflows%2Eyml/raw?ref=master"
```

I believe that OBS should be using the GitLab API to download the workflow file, not direct raw public file links.

(OBS is using the GitLab API for other things like the status reporting, as it has GitLab api token already. https://github.com/openSUSE/open-build-service/blob/a7cc5d54530b5c343c4aca250441187607db0382/src/api/app/services/gitlab_status_reporter.rb#L12 so I assume it should be able to similar use the API for fetching the workflow file)

Expected Result
============
OBS finds the workflow file for a private GitLab.com repository

Further Information
===============
* OBS server version: 2.11~alpha.20230328T133811.7ffd60ad5-150400.15356.1

cc: @dmarcoux

**Comment by krauselukas** [comment]
@nickbroon Yes, currently private repositories for Gitlab are not supported by the SCM/CI integration. I will attach the feature label to this issue. It wasn't overlooked, but we simply don't have the implementation/support for this. We also need to consider a few other factors involved to fully support private repositories.

**Comment by nickbroon** [comment]
https://www.rubydoc.info/gems/gitlab/Gitlab/Client/RepositoryFiles#file_contents-instance_method is available to fetch file contents using the GitLab API, so hopefully it shouldn't be too much to have OBS use that to get the workflow file in the same way it currently uses https://www.rubydoc.info/gems/gitlab/Gitlab%2FClient%2FCommits:update_commit_status to update commit status.

**Comment by eduardoj** [comment]
Fixed by #14641. Closing.

@nickbroon, please reopen if the issue is still not fixed.
